submissions/ses3332002/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,97 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Being {
+  constructor (name, friends, species) {
+    this.species = species;
+    this.name = name;
+    this.friends = friends;
+  }
+  
+  getDescription() {
+    return Object.entries(this)
+    .filter(([key]) => propsToPrint.includes(key))
+    .map(([,value]) => Array.isArray(value) ? value.join(', ') : value)
+    .join('; ');
+  }
+};
+
+class Animal extends Being {
+  constructor (name, friends, gender, saying, species) {
+    super(name, friends, species);
+    this.gender = gender;
+    this.saying = saying;
+    this.legs = 4;
```

Comment:
Not all animals in this world have 4 legs.
Why treating this property not the same way as e.g. species?
Besides all beings have legs just like they all have species, name and friends?
Why `legs` is a property of `Animal` and not of `Being`?

Reply:
Well, a fish or a snail, or a snake are beings, but they don't have legs. And they are not animals also.

Reply:
We do not have any snails or snakes here.
We either add them or build the system in accordance with the requirements.
Adding functionality for what is not in place yet is called premature optimization and is an anti-pattern.

Reply:
I made my world structure simpler by removing the class Being. Now the top parent class is Animal.